### PR TITLE
DockerDateFormat: trim nanonseconds for all timezones, not just Z

### LINKED ISSUE #450: Date parsing failure
## Description

We have been using 4.0.2, trying to upgrade to 5.0.1 we hit this exception when querying /info. This is happening with both docker 1.7 and 1.11. Then downgrading to 4.0.8 as thought this was a change in the 5.x branch only we hit the same issue.

Suspecting this could be related to this commit?

https://github.com/spotify/docker-client/commit/49341c70949894be8dc27690e6a7d8ff5ed40cda

But this has been there for some time, so wondering if this hasn't been an issue for anyone else.

## How to reproduce

Call DockerClient::info

## What do you expect

[Describe what do you expect to happen]

## What happened instead

Exception:

Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: Can not construct instance of java.util.Date from String value '2016-06-03T06:57:17.4782869-04:0
0': not a valid representation (error: Failed to parse Date value '2016-06-03T06:57:17.4782869-04:00': Can not parse date "2016-06-03T06:57:17.4782869-04:00": not com
patible with any of standard forms ("yyyy-MM-dd'T'HH:mm:ss.SSSZ", "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", "EEE, dd MMM yyyy HH:mm:ss zzz", "yyyy-MM-dd"))
 at [Source: org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream@6f7ec518; line: 1, column: 756] (through reference chain: com.spot
ify.docker.client.messages.Info["SystemTime"])
        at com.fasterxml.jackson.databind.exc.InvalidFormatException.from(InvalidFormatException.java:55)
        at com.fasterxml.jackson.databind.DeserializationContext.weirdStringException(DeserializationContext.java:907)
        at com.fasterxml.jackson.databind.deser.std.StdDeserializer._parseDate(StdDeserializer.java:787)

## Software:

- docker version: 
1.11.1-cs1

as well as:

1.7.1

- docker-client version: [Add docker-client version here]

### PR DESCRIPTION
**Pull request by mattnworb**
Trim the date string from nanosecond precision to millisecond precision for
all timezones, not just those ending in a literal 'Z' (for UTC).

Fixes #450.

### CODE REVIEW

**Comment by mattnworb** [comment]
@sayarg

**Comment by sayarg** [comment]
hi @mattnworb thanks vm for the quick fix. is there any way for me to try it out before a full release is published? is there a way to pull a snapshot or an rc build from maven central?

**Comment by mattnworb** [comment]
@sayarg you could always build the jar yourself from the branch

**Comment by sayarg** [comment]
yes will try that. it could be useful to have nightly snapshot pushed to maven for easier feedback loop I think but not sure how hard it may be to setup.

**Comment by mattnworb** [comment]
@davidxia

File: src/main/java/com/spotify/docker/client/DockerDateFormat.java [hunk 1]
```diff
@@ -38,14 +38,20 @@
...
+
   @Override
   public Date parse(String source) throws ParseException {
     // If the date has nanosecond precision (e.g. 2014-10-17T21:22:56.949763914Z), remove the last
     // digits so we can create a Date object, which only support milliseconds.
     // Docker doesn't always return nine digits for the fractional seconds part,
     // so we need to be more flexible when trimming to milliseconds.
-    if (source.matches(".+\\.\\d{4,9}Z$")) {
-      source = source.replaceAll("(\\.\\d{3})\\d{1,6}Z$", "$1Z");
+    // Also allow for flexible timezone formats, either 'Z' for zulu/UTC or hour offsets.
```
Comment (davidxia) [inline src/main/java/com/spotify/docker/client/DockerDateFormat.java]: TIL zulu/UTC: https://en.wikipedia.org/wiki/Coordinated_Universal_Time#Time_zones

**Comment by davidxia** [comment]
awesome 👍

**Pull request merged**